src/test/java/com/evanlennick/retry4j/CallExecutorTest_ListenersTest.java

```diff
@@ -193,6 +193,13 @@ public void verifyOnFailureListener_populatesException() {
         verify(dummyMock, timeout(1000)).listenersCallThis(isA(IllegalArgumentException.class));
     }
 
+    @Test
+    public void verifyOnSuccessListener_resultHasTypeOfCallExecutor() {
+        executor.onSuccess(status -> {
+                    String result = status.getResult(); //compile time check
```

Comment:
I generally prefer if the test assertion indicates what's being tested as opposed to some code + a comment (when possible)... what about changing this line to be `assertThat(status.getResult()).isOfAnyClassIn(String.class);`?

Additionally, if you want to be entirely thorough you should add one more test that also verifies the `AsyncCallExecutor` is also retaining the type of the result properly.